frontend/src/components/widget/RawWidget.js

```diff
@@ -352,14 +360,27 @@ export class RawWidget extends Component {
       handleProcess(caption, buttonProcessId, tabId, rowId, dataId, windowType);
   };
 
+  setWidgetType = (type) => {
+    this.rawWidget.type = type;
+  };
+
   /**
-   * @method handleErrorPopup
-   * @summary ToDo: Describe the method.
-   * @param {*} value
+   * @method showErrorPopup
+   * @summary shows error message on mouse over
+   */
```

Comment:
can't this be retained in the redux store ?

Reply:
of what ? We decided not to have widgets in the store as separate instances so we don't have a place to keep this.